src/jtag3.c

```diff
@@ -2153,6 +2171,13 @@ static int jtag3_read_byte(const PROGRAMMER *pgm, const AVRPART *p, const AVRMEM
     cmd[3] = MTYPE_OSCCAL_BYTE;
     if (pgm->flag & PGM_FL_IS_DW)
       unsupp = 1;
+  } else if (str_eq(mem->desc, "sib")) {
+    if(!*PDATA(pgm)->sib_string) {
+      pmsg_error("cannot read byte from %s sib as memory not initialised\n", p->desc);
+      return -1;
+    }
+    *value = PDATA(pgm)->sib_string[addr];
```

Comment:
Misses an addr guard against AVR_SIBLEN, like so 
```
  if(addr >= AVR_SIBLEN) {
    pmsg_error("cannot read byte from %s sib as address 0x%04lx outside range [0, 0x%04x]\n",
      p->desc, addr, AVR_SIBLEN-1);
    return -1;
  }
```

This in case a user overwrites the size of sib in avrdude.conf with 10000 and uses 9999 as address.

Reply:
wouldn't this be prevented by

```c
  if(mem->size < 1) {
    pmsg_error("cannot read byte from %s %s owing to its size %d\n", p->desc, mem->desc, mem->size);
    return -1;
  } else if(addr >= (unsigned long) mem->size) {
    pmsg_error("cannot read byte from %s %s as address 0x%04lx outside range [0, 0x%04x]\n",
      p->desc, mem->desc, addr, mem->size-1);
    return -1;
  }
```
 at the beginning of jtag3_read_byte?